src/main.ts

```diff
@@ -35,14 +36,18 @@ export const main = async ({
 
     const url = new URL(GITLAB_HOST)
 
-    await exec('git', [
-      'remote',
-      'set-url',
-      'origin',
-      `${url.protocol}//${GITLAB_CI_USER_NAME!}:${GITLAB_TOKEN!}@${
-        url.host
-      }/${CI_PROJECT_PATH!}.git`,
-    ])
+    await exec(
+      'git',
+      [
+        'remote',
+        'set-url',
+        'origin',
+        `${url.protocol}//${GITLAB_CI_USER_NAME!}:${GITLAB_TOKEN!}@${
+          url.host
+        }/${CI_PROJECT_PATH!}.git`,
+      ],
+      { silent: !['true', '1'].includes(DEBUG_GITLAB_CREDENTIAL) },
```

Comment:
```suggestion
      { silent: !['true', '1'].includes(DEBUG_GITLAB_CREDENTIAL!) },
```

Sorry the types are a bit confusing.